src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -379,7 +379,7 @@ RegMask _STACK_OR_LONG_REG_mask;
 RegMask _STACK_OR_INT_REG_mask;
 
 static bool need_r12_heapbase() {
-  return UseCompressedOops;
+  return CompressedOops::need_heapbase_reg();
```

Comment:
Why not just use `PreserveHeapbaseReg`, this is a machine-dependent flag so I don't find it appealing to introduce a machine-independent check.